bundles/automation/org.eclipse.smarthome.automation.core/src/main/java/org/eclipse/smarthome/automation/core/internal/RuleRegistryImpl.java

```diff
@@ -251,7 +255,7 @@ public Rule add(Rule rule) {
      * @param rule candidate for unique ID
      * @return a rule with UID
      */
-    protected Rule initRuleId(String rUID, Rule rule) {
+    protected @Nonnull Rule initRuleId(String rUID, Rule rule) {
```

Comment:
Is than suitable to remove null pointer checks in all places in this commit? For example in the AbstractManagedProvider.add(E elemetn) was removed 
        if (element == null) {
            throw new IllegalArgumentException("Cannot add null element");
        }
which will lead to the NPE instead of exception with describable message